### PR TITLE
Bedrock Embedding support (duplicate)

### PR DESCRIPTION
**Pull request by ryanh-ai**
@ishaan-jaff when I started working on support for Bedrock Embedding, you hadn't committed your similar commit ( #701 ,  Commit ID 2ad81bd ). Primary differences I see at first glance:

- you have some type/error checking I don't have to validate the input is a list of length 1 and elements are type string
- I enabled input list length > 1 by spawning multiple calls to bedrock API
- I added logging support modeled after your Cohere embedding support

Please take a look.  If you want to keep some portion of this, I'll think about how to add async support to the single call to Bedrock so those calls aren't blocking.

I DID not resolve conflicts here as your approach was slightly different.  I can if you want to weigh in here, or simply reject and I'll rebase.

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
@canada4663 thanks for the contribution - will take a look at this later and respond in 24 hours

File: litellm/llms/bedrock.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def init_bedrock_client(
     elif standard_aws_region_name:
         region_name = standard_aws_region_name
     else:
-        raise BedrockError(message="AWS region not set: set AWS_REGION_NAME or AWS_REGION env variable or in .env file")
+        raise BedrockError(message="AWS region not set: set AWS_REGION_NAME or AWS_REGION env variable or in .env file", status_code=401)
```
Comment (ishaan-jaff) [inline litellm/llms/bedrock.py]: we need this `raise BedrockError(message="AWS region not set: set AWS_REGION_NAME or AWS_REGION env variable or in .env file", status_code=401)`

File: litellm/main.py [hunk 2]
```diff
@@ -1632,6 +1633,16 @@ def embedding(
                 logging_obj=logging,
                 model_response= EmbeddingResponse()
             )
+        elif custom_llm_provider == "bedrock":
```
Comment (ishaan-jaff) [inline litellm/main.py]: this is duplicate code

File: litellm/tests/test_embedding.py [hunk 3]
```diff
@@ -78,6 +78,17 @@ def test_cohere_embedding():
 
 # test_cohere_embedding()
 
+def test_bedrock_embedding():
```
Comment (ishaan-jaff) [inline litellm/tests/test_embedding.py:95]: duplicate code here too

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]

**Comment by ishaan-jaff** [comment]
@canada4663 thanks for the contribution ! 
- there was a lot of duplicate code in this PR - we already have most of this functionality 

Can you deal with merge conflicts ? happy to take a look at this PR after that

**Comment by ishaan-jaff** [comment]
> I enabled input list length > 1 by spawning multiple calls to bedrock API

I like this - so keen to see if we can get this merged

**Comment by ryanh-ai** [comment]
sounds good, will take a look this weekend.

File: litellm/llms/bedrock.py (hunk 1 above)
Comment (ryanh-ai) [reply litellm/llms/bedrock.py]: merged upstream in

File: litellm/main.py (hunk 2 above)
Comment (ryanh-ai) [reply litellm/main.py]: eliminated

File: litellm/tests/test_embedding.py (hunk 3 above)
Comment (ryanh-ai) [reply litellm/tests/test_embedding.py:95]: eliminated

**Comment by canada4663** [comment]
@ishaan-jaff  - I merged in everything upstream to eliminate the duplicates you called out, and added in the capability of accepting >1 string to be embedded at a time.

It won't let me re-open the PR, so if you see the diff and are OK I can open a fresh with my changes.

**Comment by ryanh-ai** [comment]
@ishaan-jaff closed this again as the commit has bunch of white space changes I want to get rid of to minimize the diff to make PR clean. Will send tomorrow.

**Pull request closed without merging**